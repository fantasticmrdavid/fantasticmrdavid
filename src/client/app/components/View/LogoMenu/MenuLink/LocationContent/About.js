import React from 'react';

export default (props) => {
  return <div>
          Meet the captain of the ship.
        </div>;
};
