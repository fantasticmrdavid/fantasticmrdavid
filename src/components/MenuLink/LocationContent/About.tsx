import React from 'react';

export default () => {
  return <div>Meet the captain of the ship.</div>;
};
