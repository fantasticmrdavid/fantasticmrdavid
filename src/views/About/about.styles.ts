import styled, { css } from 'styled-components';
import theme from 'styles/theme';
import { transparentize } from 'polished';
import { fadeIn } from 'styles/animations';
import { media } from 'styles/utils';
import ComponentLink from 'components/Link';

interface ContainerProps {
  isLoading: boolean,
}

export const Container = styled.div<ContainerProps>`
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  box-sizing: border-box;
  min-height: 100%;
  padding: 2em;
  background-color: ${theme.colorDarkCharcoal};
  color: ${theme.colorTextPrimary};
  opacity: ${(props) => (props.isLoading ? 0 : 1)};
  transition: 1s;
  animation: ${css`${fadeIn} 1s ease 1`};
  animation-fill-mode: forwards;

  ${media.small`
    padding: 4em 2em;
    flex-direction: row-reverse;
    align-items: flex-start;
  `}
`;

export const Bg = styled.div`
  position: absolute;
  top: 0px;
  right: 0px;
  width: 100%;
  height: 100%;
  overflow: hidden;
`;

export const ProfilePhoto = styled.img`
  width: 100px;
  height: auto;
  margin-bottom: 1em;
  z-index: 1;

  ${media.small`
    margin-bottom: 0;
  `};
`;

export const Content = styled.div`
  flex: 1;
  font-size: 0.9rem;
  font-weight: 300;
  line-height: 1.3;
  max-width: 100%;
  padding-right: 0px;
  z-index: 1;

  ${media.small`
    padding-right: 1em;
    max-width: 900px;
  `}

  h2{
    color: ${theme.colorWhite};
  }

  p {
    margin-bottom: 1em;
  }

  ul {
    list-style: disc;
    margin-top: 0.25em;
    padding-left: 1.2em;
  }
`;

export const SiteSpecs = styled.div`
  display: grid;
  grid-template-columns: 1fr 1fr;

  ${media.large`
    grid-template-columns: 1fr 1fr 1fr 1fr;
  `}
`;

export const SpecsSection = styled.div`
  padding: 1em 0;

  h3 {
    color: ${theme.colorWhite};
  }
`;

export const SleeperLink = styled(ComponentLink)`
  color: ${theme.colorTextPrimary};
  border-bottom: ${`1px solid ${transparentize(0.8, theme.colorTextPrimary)}`};

  ${media.hover`
    &:hover {
      color: ${theme.colorWhite};
    }
  `};
`;
