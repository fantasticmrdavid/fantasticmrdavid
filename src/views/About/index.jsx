import React from 'react';
import PropTypes from 'prop-types';
import Cube from 'components/Cube';
import * as styles from './styles';

const AboutMedia = ({ loading }) => {
  const {
    Container,
    Bg,
    ProfilePhoto,
    Content,
    SleeperLink,
    SiteSpecs,
    SpecsSection,
  } = styles;
  return (
    <Container loading={loading}>
      <Bg><Cube loading={loading} /></Bg>
      <ProfilePhoto src="/assets/images/profile.jpg" alt="Me" />
      <Content>
        <p>
          Hi! I'm David, and I like making things.<br />
          Since childhood I've been passionate about building and creating.
          I've always enjoyed the journey of taking an idea from concept and
          making it reality whether it's solving a complex problem, creating some
          ambitious new artwork, or both.
        </p>

        <h2>Engineer</h2>
        <p>
          I've been building web projects for over a decade, starting from
          when I was in high school and needed a simple HTML page to display my
          artwork. From there I moved on to become a Full-Stack LAMP Developer
          working on vendor-side and agency projects, and then to
          lead Front-End and UX Engineer specialising in HTML5, CSS3, ES6 and React. I
          love learning about new technologies to develop better and more effective
          product solutions.
        </p>

        <h2>Artist</h2>
        <p>
          Drawing has been my happy place for as long as I can remember.
          While building web apps keeps the problem solving part of my brain busy,
          illustration and artwork have allowed me to stretch my creative boundaries.
          In my after hours I have drawn comics, exhibited artwork, produced 3D
          prints, and made <SleeperLink href="https://www.youtube.com/watch?v=7xk4LmZ6hwc" target="_blank" rel="noopener noreferrer">other random things</SleeperLink> simply because they were too much fun.
        </p>

        <h2>About this site</h2>
        fantasticmrdavid.com.au was built with:
        <SiteSpecs>
          <SpecsSection>
            <h3>Environment</h3>
            <ul>
              <li>ES6</li>
              <li>Node.js</li>
              <li>Webpack</li>
            </ul>
          </SpecsSection>
          <SpecsSection>
            <h3>Architecture</h3>
            <ul>
              <li>React</li>
              <li>Redux</li>
              <li>CSS-Modules</li>
            </ul>
          </SpecsSection>
          <SpecsSection>
            <h3>Testing/QA</h3>
            <ul>
              <li>ESLint</li>
              <li>Jest</li>
            </ul>
          </SpecsSection>
          <SpecsSection>
            <h3>Deployment</h3>
            <ul>
              <li>Github</li>
              <li>Codeship</li>
              <li>AWS</li>
            </ul>
          </SpecsSection>
        </SiteSpecs>
        <p>Disable JavaScript to enable the noscript version!</p>
        <a className="cta" href="https://github.com/fantasticmrdavid/fantasticmrdavid" target="_blank" rel="noopener noreferrer">
        Visit the Github repository
        </a>
      </Content>
    </Container>
  );
};

AboutMedia.propTypes = {
  loading: PropTypes.bool,
};

export default AboutMedia;
