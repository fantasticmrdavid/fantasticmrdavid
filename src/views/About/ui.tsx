import React, { memo, useContext } from 'react';
import Cube from 'components/Cube';
import Link from 'components/Link';
import { LoadingContext } from 'contexts/Loading';
import {
  Container,
  Bg,
  ProfilePhoto,
  Content,
  SleeperLink,
  SiteSpecs,
  SpecsSection,
} from './styles';

export default memo(
  () => {
    const { getIsLoading } = useContext(LoadingContext);
    const isLoading = getIsLoading();
    return (
      <Container isLoading={isLoading}>
        <Bg><Cube isLoading={isLoading} /></Bg>
        <ProfilePhoto src="/assets/images/profile.jpg" alt="Me" />
        <Content>
          <p>
            Hi! I'm David, and I like making things.
            <br /><br />
            Since I was a kid I've been passionate about building and creating.
            I've always enjoyed the journey of taking an idea from concept and
            making it reality.
            <br /><br />
            I firmly believe that by empowering people with problem solving technology
            and empathy we can build a better world together.
          </p>

          <h2>Engineer</h2>
          <p>
            I've been building web projects for over a decade, starting from
            when I was in high school and needed a simple HTML page to display my
            artwork. From there my journey has taken me to Full-Stack LAMP Development
            working on vendor-side and agency projects alike, and then to leading
            Front-End and UX Engineering specialising in HTML5, CSS3, ES6 JavaScript and React.
            I love learning about new technologies to develop better and more effective
            ways to solve problems and get things done.
          </p>

          <h2>Artist</h2>
          <p>
            Drawing has been my happy place for as long as I can remember.
            While building software keeps the puzzle solving part of my brain busy,
            illustration, artwork, and tinkering have allowed me to stretch my creative boundaries.
            In my downtime I've drawn comics, exhibited artwork, designed and produced 3D
            prints, and made <SleeperLink href="https://www.youtube.com/watch?v=7xk4LmZ6hwc" target="_blank" rel="noopener noreferrer">other random things</SleeperLink> simply because they were too much fun.
          </p>

          <h2>About this site</h2>
          fantasticmrdavid.com.au was built with:
          <SiteSpecs>
            <SpecsSection>
              <h3>Environment</h3>
              <ul>
                <li>ES6</li>
                <li>Node.js</li>
                <li>Webpack</li>
              </ul>
            </SpecsSection>
            <SpecsSection>
              <h3>Architecture</h3>
              <ul>
                <li>React</li>
                <li>Redux</li>
                <li>Styled-Components</li>
              </ul>
            </SpecsSection>
            <SpecsSection>
              <h3>Testing/QA</h3>
              <ul>
                <li>TypeScript</li>
                <li>ESLint</li>
                <li>Jest</li>
                <li>Enzyme</li>
              </ul>
            </SpecsSection>
            <SpecsSection>
              <h3>Deployment</h3>
              <ul>
                <li>Github</li>
                <li>Codeship</li>
                <li>AWS</li>
              </ul>
            </SpecsSection>
          </SiteSpecs>
          <p>Disable JavaScript to enable the noscript version!</p>
          <Link href="https://github.com/fantasticmrdavid/fantasticmrdavid" target="_blank" rel="noopener noreferrer">
            Visit the Github repository
          </Link>
        </Content>
      </Container>
    );
  },
);
