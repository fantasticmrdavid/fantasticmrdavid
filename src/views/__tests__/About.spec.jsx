import React from 'react';
import { mount } from 'enzyme';
import About from 'views/About';

describe('View -> About', () => {
  it('should render the About view (loading) without throwing an error', () => {
    expect(mount(<About loading />)).toMatchSnapshot();
  });

  it('should render the About view (not loading) without throwing an error', () => {
    expect(mount(<About />)).toMatchSnapshot();
  });
});
