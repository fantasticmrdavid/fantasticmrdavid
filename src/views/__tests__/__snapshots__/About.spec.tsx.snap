// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`View -> About should render the About view without throwing an error 1`] = `
{
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="styles__Container-sc-qm8p7s-0 eoAhMa"
      >
        <div
          class="styles__Bg-sc-qm8p7s-1 bPLvTD"
        >
          <div
            class="styles__Container-sc-1pwogyu-0 fkagAV"
          >
            <div
              class="styles__Polygon-sc-1pwogyu-1 dCIqjj"
            >
              <div
                class="styles__Side-sc-1pwogyu-2 styles__Front-sc-1pwogyu-8 deBIOm"
              >
                <div
                  class="styles__Tracer-sc-18vfsm6-0 hNwUQp"
                  direction="left"
                />
              </div>
              <div
                class="styles__Side-sc-1pwogyu-2 styles__Back-sc-1pwogyu-3 gCcrQb"
              >
                <div
                  class="styles__Tracer-sc-18vfsm6-0 lKbcB"
                />
              </div>
              <div
                class="styles__Side-sc-1pwogyu-2 styles__Top-sc-1pwogyu-6 eLxrDv"
              />
              <div
                class="styles__Side-sc-1pwogyu-2 styles__Bottom-sc-1pwogyu-7 cpagcj"
              >
                <div
                  class="styles__Tracer-sc-18vfsm6-0 lKbcB"
                />
              </div>
              <div
                class="styles__Side-sc-1pwogyu-2 styles__Left-sc-1pwogyu-5 iDSTJS"
              >
                <div
                  class="styles__Tracer-sc-18vfsm6-0 hNwUQp"
                  direction="left"
                />
              </div>
              <div
                class="styles__Side-sc-1pwogyu-2 styles__Right-sc-1pwogyu-4 fTRfub"
              >
                <div
                  class="styles__Tracer-sc-18vfsm6-0 lKbcB"
                />
              </div>
            </div>
          </div>
        </div>
        <img
          alt="Me"
          class="styles__ProfilePhoto-sc-qm8p7s-2 cToiw"
          src="/assets/images/profile.jpg"
        />
        <div
          class="styles__Content-sc-qm8p7s-3 ciksqE"
        >
          <p>
            Hi! I'm David, and I like making things.
            <br />
            <br />
            Since I was a kid I've been passionate about building and creating. I've always enjoyed the journey of taking an idea from concept and making it reality.
            <br />
            <br />
            I firmly believe that by empowering people with problem solving technology and empathy we can build a better world together.
          </p>
          <h2>
            Engineer
          </h2>
          <p>
            I've been building web projects for over a decade, starting from when I was in high school and needed a simple HTML page to display my artwork. From there my journey has taken me to Full-Stack LAMP Development working on vendor-side and agency projects alike, and then to leading Front-End and UX Engineering specialising in HTML5, CSS3, ES6 JavaScript and React. I love learning about new technologies to develop better and more effective ways to solve problems and get things done.
          </p>
          <h2>
            Artist
          </h2>
          <p>
            Drawing has been my happy place for as long as I can remember. While building software keeps the puzzle solving part of my brain busy, illustration, artwork, and tinkering have allowed me to stretch my creative boundaries. In my downtime I've drawn comics, exhibited artwork, designed and produced 3D prints, and made 
            <a
              class="styles__Link-sc-1rd8ebj-0 jONwCq styles__SleeperLink-sc-qm8p7s-6 cFShCR"
              href="https://www.youtube.com/watch?v=7xk4LmZ6hwc"
              rel="noopener noreferrer"
              target="_blank"
            >
              other random things
            </a>
             simply because they were too much fun.
          </p>
          <h2>
            About this site
          </h2>
          fantasticmrdavid.com.au was built with:
          <div
            class="styles__SiteSpecs-sc-qm8p7s-4 hIVtxr"
          >
            <div
              class="styles__SpecsSection-sc-qm8p7s-5 dkQZmC"
            >
              <h3>
                Environment
              </h3>
              <ul>
                <li>
                  ES6
                </li>
                <li>
                  Node.js
                </li>
                <li>
                  Webpack
                </li>
              </ul>
            </div>
            <div
              class="styles__SpecsSection-sc-qm8p7s-5 dkQZmC"
            >
              <h3>
                Architecture
              </h3>
              <ul>
                <li>
                  React
                </li>
                <li>
                  Styled-Components
                </li>
              </ul>
            </div>
            <div
              class="styles__SpecsSection-sc-qm8p7s-5 dkQZmC"
            >
              <h3>
                Testing/QA
              </h3>
              <ul>
                <li>
                  TypeScript
                </li>
                <li>
                  ESLint
                </li>
                <li>
                  Prettier
                </li>
                <li>
                  Jest
                </li>
                <li>
                  Testing Library
                </li>
              </ul>
            </div>
            <div
              class="styles__SpecsSection-sc-qm8p7s-5 dkQZmC"
            >
              <h3>
                Deployment
              </h3>
              <ul>
                <li>
                  Github
                </li>
                <li>
                  Codeship
                </li>
                <li>
                  AWS
                </li>
              </ul>
            </div>
          </div>
          <p>
            Disable JavaScript to enable the noscript version!
          </p>
          <a
            class="styles__Link-sc-1rd8ebj-0 jONwCq"
            href="https://github.com/fantasticmrdavid/fantasticmrdavid"
            rel="noopener noreferrer"
            target="_blank"
          >
            Visit the Github repository
          </a>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="styles__Container-sc-qm8p7s-0 eoAhMa"
    >
      <div
        class="styles__Bg-sc-qm8p7s-1 bPLvTD"
      >
        <div
          class="styles__Container-sc-1pwogyu-0 fkagAV"
        >
          <div
            class="styles__Polygon-sc-1pwogyu-1 dCIqjj"
          >
            <div
              class="styles__Side-sc-1pwogyu-2 styles__Front-sc-1pwogyu-8 deBIOm"
            >
              <div
                class="styles__Tracer-sc-18vfsm6-0 hNwUQp"
                direction="left"
              />
            </div>
            <div
              class="styles__Side-sc-1pwogyu-2 styles__Back-sc-1pwogyu-3 gCcrQb"
            >
              <div
                class="styles__Tracer-sc-18vfsm6-0 lKbcB"
              />
            </div>
            <div
              class="styles__Side-sc-1pwogyu-2 styles__Top-sc-1pwogyu-6 eLxrDv"
            />
            <div
              class="styles__Side-sc-1pwogyu-2 styles__Bottom-sc-1pwogyu-7 cpagcj"
            >
              <div
                class="styles__Tracer-sc-18vfsm6-0 lKbcB"
              />
            </div>
            <div
              class="styles__Side-sc-1pwogyu-2 styles__Left-sc-1pwogyu-5 iDSTJS"
            >
              <div
                class="styles__Tracer-sc-18vfsm6-0 hNwUQp"
                direction="left"
              />
            </div>
            <div
              class="styles__Side-sc-1pwogyu-2 styles__Right-sc-1pwogyu-4 fTRfub"
            >
              <div
                class="styles__Tracer-sc-18vfsm6-0 lKbcB"
              />
            </div>
          </div>
        </div>
      </div>
      <img
        alt="Me"
        class="styles__ProfilePhoto-sc-qm8p7s-2 cToiw"
        src="/assets/images/profile.jpg"
      />
      <div
        class="styles__Content-sc-qm8p7s-3 ciksqE"
      >
        <p>
          Hi! I'm David, and I like making things.
          <br />
          <br />
          Since I was a kid I've been passionate about building and creating. I've always enjoyed the journey of taking an idea from concept and making it reality.
          <br />
          <br />
          I firmly believe that by empowering people with problem solving technology and empathy we can build a better world together.
        </p>
        <h2>
          Engineer
        </h2>
        <p>
          I've been building web projects for over a decade, starting from when I was in high school and needed a simple HTML page to display my artwork. From there my journey has taken me to Full-Stack LAMP Development working on vendor-side and agency projects alike, and then to leading Front-End and UX Engineering specialising in HTML5, CSS3, ES6 JavaScript and React. I love learning about new technologies to develop better and more effective ways to solve problems and get things done.
        </p>
        <h2>
          Artist
        </h2>
        <p>
          Drawing has been my happy place for as long as I can remember. While building software keeps the puzzle solving part of my brain busy, illustration, artwork, and tinkering have allowed me to stretch my creative boundaries. In my downtime I've drawn comics, exhibited artwork, designed and produced 3D prints, and made 
          <a
            class="styles__Link-sc-1rd8ebj-0 jONwCq styles__SleeperLink-sc-qm8p7s-6 cFShCR"
            href="https://www.youtube.com/watch?v=7xk4LmZ6hwc"
            rel="noopener noreferrer"
            target="_blank"
          >
            other random things
          </a>
           simply because they were too much fun.
        </p>
        <h2>
          About this site
        </h2>
        fantasticmrdavid.com.au was built with:
        <div
          class="styles__SiteSpecs-sc-qm8p7s-4 hIVtxr"
        >
          <div
            class="styles__SpecsSection-sc-qm8p7s-5 dkQZmC"
          >
            <h3>
              Environment
            </h3>
            <ul>
              <li>
                ES6
              </li>
              <li>
                Node.js
              </li>
              <li>
                Webpack
              </li>
            </ul>
          </div>
          <div
            class="styles__SpecsSection-sc-qm8p7s-5 dkQZmC"
          >
            <h3>
              Architecture
            </h3>
            <ul>
              <li>
                React
              </li>
              <li>
                Styled-Components
              </li>
            </ul>
          </div>
          <div
            class="styles__SpecsSection-sc-qm8p7s-5 dkQZmC"
          >
            <h3>
              Testing/QA
            </h3>
            <ul>
              <li>
                TypeScript
              </li>
              <li>
                ESLint
              </li>
              <li>
                Prettier
              </li>
              <li>
                Jest
              </li>
              <li>
                Testing Library
              </li>
            </ul>
          </div>
          <div
            class="styles__SpecsSection-sc-qm8p7s-5 dkQZmC"
          >
            <h3>
              Deployment
            </h3>
            <ul>
              <li>
                Github
              </li>
              <li>
                Codeship
              </li>
              <li>
                AWS
              </li>
            </ul>
          </div>
        </div>
        <p>
          Disable JavaScript to enable the noscript version!
        </p>
        <a
          class="styles__Link-sc-1rd8ebj-0 jONwCq"
          href="https://github.com/fantasticmrdavid/fantasticmrdavid"
          rel="noopener noreferrer"
          target="_blank"
        >
          Visit the Github repository
        </a>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
